File: carddemo-java/src/main/java/com/aws/carddemo/model/CardXref.java
package com.aws.carddemo.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

/**
 * Card Cross-Reference entity - migrated from COBOL copybook CVACT03Y.cpy
 * Original COBOL record length: 50 bytes
 * Links card numbers to customer and account IDs
 */
@Entity
@Table(name = "CARDXREF")
public class CardXref {

    @Id
    @Column(name = "XREF_CARD_NUM", length = 16)
    private String xrefCardNum;

    @Column(name = "XREF_CUST_ID")
    private Long xrefCustId;

    @Column(name = "XREF_ACCT_ID")
    private Long xrefAcctId;

    public CardXref() {
    }

    public CardXref(String xrefCardNum) {
        this.xrefCardNum = xrefCardNum;
    }

    public String getXrefCardNum() {
        return xrefCardNum;
    }

    public void setXrefCardNum(String xrefCardNum) {
        this.xrefCardNum = xrefCardNum;
    }

    public Long getXrefCustId() {
        return xrefCustId;
    }

    public void setXrefCustId(Long xrefCustId) {
        this.xrefCustId = xrefCustId;
    }

    public Long getXrefAcctId() {
        return xrefAcctId;
    }

    public void setXrefAcctId(Long xrefAcctId) {
        this.xrefAcctId = xrefAcctId;
    }

    @Override
    public String toString() {
        return "CardXref{" +
                "xrefCardNum='" + xrefCardNum + '\'' +
                ", xrefCustId=" + xrefCustId +
                ", xrefAcctId=" + xrefAcctId +
                '}';
    }
}

File: carddemo-batch-python/src/carddemo_batch/models/card_xref.py
"""CardXref model - migrated from COBOL copybook CVACT03Y.cpy (50 bytes)."""

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column

from carddemo_batch.models.base import Base


class CardXref(Base):
    """
    Card cross-reference entity representing the CARDXREF VSAM file.
    
    Migrated from COBOL copybook CVACT03Y.cpy.
    Record length: 50 bytes
    Primary key: XREF-CARD-NUM (16 characters)
    Links card to customer and account.
    """
    __tablename__ = "cardxref"

    xref_card_num: Mapped[str] = mapped_column(String(16), primary_key=True)
    xref_cust_id: Mapped[int] = mapped_column(nullable=True)
    xref_acct_id: Mapped[int] = mapped_column(nullable=True)

    def __repr__(self) -> str:
        return f"CardXref(card={self.xref_card_num}, cust={self.xref_cust_id}, acct={self.xref_acct_id})"
